### PR TITLE
[Parse] Disallow unsupported condition expression in #if directive

### PR DESCRIPTION
**Pull request by rintaro**
In non-Swift3 mode.

Previously:

```swift
#if FOO = false
#elseif FOO ? false : true
#endif
```
were silently accepted.
i.e. `= false` and `? false : true` were silently ignored.

In Swift3 mode, emit a warning:

```
warning: ignoring invalid conditional compilation expression, which will be rejected in future version of Swift
#if FOO ? BAR : BAZ && QUX
        ^~~~~~~~~~~
```

### CODE REVIEW

File: lib/Parse/ParseStmt.cpp [hunk 1]
```diff
@@ -1570,10 +1570,22 @@ Parser::classifyConditionalCompilationExpr(Expr *condition,
...
+      } else {
+        // Swift3 didn't have this branch. the operator and the RHS are
+        // silently ignored.
+        if (!Context.isSwiftVersion3()) {
+          D.diagnose(
+              elements[iOperator]->getLoc(),
+              diag::unsupported_conditional_compilation_expression_type);
+          return ConditionalCompilationExprState::error();
+        } else {
+          // TODO: Emit a warning.
```
Comment (rintaro) [inline lib/Parse/ParseStmt.cpp]: Any suggestion for the message?

```swift
#if a ? b : 12
```

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: include/swift/AST/DiagnosticsParse.def [hunk 2]
```diff
@@ -1356,6 +1356,9 @@ ERROR(unexpected_version_comparison_operator,none,
       ())
 ERROR(unsupported_conditional_compilation_expression_type,none,
       "invalid conditional compilation expression", ())
+WARNING(swift3_unsupported_conditional_compilation_expression_type,none,
```
Comment (slavapestov) [inline include/swift/AST/DiagnosticsParse.def]: Nitpick: align whitespace on next line with (

**Comment by rintaro** [comment]
@swift-ci Please smoke test?

**Pull request merged**